Respect original remember-me cookie on impersonator's login after impersonation stops

j3j5 opened the pull request:
It fixes #6.

Right now phpstan is complaining that:

> Call to function is_callable() with array{Illuminate\Contracts\Auth\Guard, 'getRequest'} will always evaluate to true.

but afaics the contract does not include a `getRequest()` method, which is why I included the checks.

Not sure how to fix it.

j3j5 committed:
chore: check whether the remember cookie was set when logging the impersonator back

franbarbalopez commented on src/Impersonator.php:
```diff
@@ -427,4 +444,24 @@ protected function dispatchEventAfterResponse(object $event): void
+            || is_null($guard->getRequest())
+        ) {
+            return null;
+        }
```
```suggestion
    protected function recaller(SessionGuard $guard): ?Recaller
    {
        if (is_null($guard->getRequest())) {
            return null;
        }
```

As the remember token only works with `SessionGuard`, I think we should type–hint the `$guard` parameter accordingly. We also don’t need to check both callables.

franbarbalopez commented on src/Impersonator.php:
```diff
@@ -158,13 +161,27 @@ protected function performStop(): void
+                $recallerId = intval($recallerId);
+            }
+
+            $remember = $impersonatorId === $recallerId;
```
I think we could simplify this by using `$impersonatorId == $recaller->id()`, and we’d also get UUID/ULID support

```suggestion
            $remember = $impersonatorId == $recaller->id();
```

franbarbalopez reviewed (changes requested):
Thank you for the PR @j3j5 !

Please check the suggested changes and tell me if you see something wrong with them.

You think you could bring any feature test? I can do this if you want.

j3j5 committed:
Replace type handling to make strict comparison with loose comparison.

The id() method from the recaller always returns a string, so the loose comparison allows to support integers, ULIDs and UUIDs. The loose comparison is going to have the same effect when comparing to string than `intval()` would have when comparing to an int.

Co-authored-by: Francisco José Barba López <franciscojose.barbalopez@gmail.com>

j3j5 committed:
Add a couple of tests for the remember cookie

j3j5 commented:
Gracias a ti por publicar y mantener el paquete 💪
